### PR TITLE
Add App: Piconacci

### PR DESCRIPTION
**Pull request by seanbechhofer**
Adding Piconacci app. Simple clock divider with division values taken from the Fibonacci series.

### CODE REVIEW

File: software/contrib/piconacci.py
Comment (atulrnt) [inline software/contrib/piconacci.py:113]: [optional] Updating the script at every cycle is quite resource heavy.
A way to save resources could be the store the state of the output in a variable and check if that variable changed before updating the display.

In a script I made, I was also refreshing the display at every cycle and the output ended up shifting in time.  
Implementing the solution I'm proposing here does help with that.

File: software/contrib/piconacci.py
Comment (atulrnt) [inline software/contrib/piconacci.py:111]: [question] I don't see this variable being used anywhere, am I missing something?

**Comment by atulrnt** [comment]
Hey @seanbechhofer , thanks for submitting this cool contribution!  
This is my first review in this project but I'd like to give it a try so please bare with me :)

If there is anything wrong with my review, we'll ask someone with more experience in this project to completely take over and whatever the outcome of my review is, it will need the approval of someone else.

I'm just trying to help a bit where I can, so let's get started!

**Review by atulrnt — CHANGES_REQUESTED** [review]

**Comment by seanbechhofer** [comment]
Thanks for the review. I've push a new set of changes to address the issues.

**Comment by atulrnt** [comment]
Thanks for working on those changes, looking all good to me, now waiting for an official contributor to review and approve :)  
@mjaskula I guess?

File: software/contrib/piconacci.py
Comment (mjaskula) [inline software/contrib/piconacci.py:44]: [optional] Consider variable names that make descriptive comments redundant, allowing you to remove or reduce them. For example instead of `dirty` you could call this something like `display_update_required`. It's a little more verbose, but the comments here and where it is used become unnecessary.

File: software/contrib/piconacci.py
Comment (mjaskula) [inline software/contrib/piconacci.py:66]: [optional] These three lines of comments don't seem to be adding value that isn't captured by the decorator. Same with the comments on `b2Pressed` below.

File: software/contrib/piconacci.py
Comment (mjaskula) [inline software/contrib/piconacci.py:118]: [optional] I don't think that this is necessary here. The initial state is `dirty = True`, so the first trip through the while loop below will update the screen.

File: software/contrib/piconacci.py
Comment (mjaskula) [inline software/contrib/piconacci.py:113]: [optional] If you were to move the code for the `value()` function into the calling `clockTrigger()` you would notice that the `values` list id being created each time through the for loop, but it doesn't need to be. I think that removing the `value()` function might actually lead to easier to understand code in `clockTrigger()`

**Review by mjaskula — APPROVED** [review]
This looks pretty good to me. The code is clean and simple. I've left a few comments of places where I think there is opportunity to improve things in my eyes. All optional. Let me know what you think ands then we can get it merged.

File: software/contrib/piconacci.py
Comment (seanbechhofer) [reply software/contrib/piconacci.py]: I'm inclined to leave the function declaration. For me, it makes things clearer. Also, it gets used in the display update. But there is no need for the intermediate variable.

**Pull request merged**
